### ISSUE TITLE
Get entity ID for File Upload

### ISSUE THREAD

**Issue by pdamicol** [marmelab/ng-admin#1144]
Hi!

I'm using the FileField and my upload endpoint is something like this:

> http://api.example.com/entity/1/

The problem is that I need to provide the URL when I'm defining the field, without knowing the entity's ID, like this:
`nga.field('logo', 'file').uploadInformation({ 'url': 'http://api.example.com/entity/ID_HERE/', 'accept': '.svg'})`

How can I provide the ID of the current entity to upload the file?

P.S. I only use this field in the editionView, therefore the ID is always available.

**Comment by jpetitcolas** [comment]
Indeed, we should be able to pass a function to `uploadInformation`, passing it the server response and entry as arguments. This way, you would be able to do something like:

``` js
nga.field('logo', 'file')
    .uploadInformation((response, entry) => ({
        url: `http://api.example.com/entity/${entry.id}`,
        accept: 'svg',
    }));
```

**Comment by mabbasian** [comment]
@jpetitcolas this does not work, I get following error
`Error: You must provide a URL property to allow the upload of files.`

**Comment by ledfusion** [comment]
Is there any plan to support dynamic information provided on `.uploadInformation()`?

**Comment by situokko** [comment]
+1, as right now the file-uploading is useless as there is no way to bind the uploaded image to the entity being edited.

**Comment by lobodormido** [comment]
I need this :(

**Comment by jessequinn** [comment]
any progress on this? Otherwise uploadInformation just seems pointless for dynamic content!!!!
